luwak/src/main/java/uk/co/flax/luwak/termextractor/querytree/DisjunctionNode.java

```diff
@@ -25,43 +28,44 @@
 
 public class DisjunctionNode extends QueryTree {
 
+    private final List<QueryTree> children = new ArrayList<>();
+
     private DisjunctionNode(List<QueryTree> children) {
-        for (QueryTree child : children) {
-            this.addChild(child);
-        }
+        this.children.addAll(children);
```

Comment:
Instead of making new ArrayList in the class body, could move the new ArrayList to within constructor = this.children = new ArrayList<>(children) which would exactly size every time.